pkg/machinery/config/types/v1alpha1/v1alpha1_validation_test.go

```diff
@@ -2069,3 +2073,158 @@ func TestValidateCNI(t *testing.T) {
 		})
 	}
 }
+
+func TestKubernetesVersionFromImageRef(t *testing.T) {
+	t.Parallel()
+
+	for _, test := range []struct {
+		imageRef string
+
+		expectedVersion string
+	}{
+		{
```

Comment:
maybe also have ones without tag and one with just sha and no tag

Reply:
we don't allow no tag, as we won't be able to guess the version this way....

Reply:
but this might be a breaking change, so I will add a note that the tag is required